Security update (TLS communications CP<->DP) (#298)
kumahq/kuma-website: docs/docs/0.7.2/documentation/security.md

# Security

There are several communication use cases that can be secured when using Kuma.

## Data plane to data plane communication

Kuma helps you secure your existing infrastructure with mTLS.

### Mutual TLS

Once a data plane has proved its identity to the control plane, it will be allowed to fetch its own identity certificate and a root CA certificate of the mesh.

> When establishing a connection between two data planes each side validates each other data plane certificate confirming the identity using the root CA of the mesh.

Reviewer: ```suggestion
When establishing a connection between two data planes, each side validates the other data plane's certificate using the root CA of the mesh, which then confirms their identity.
```

mTLS is _not_ enabled by default. To enable it, apply proper settings in [Mesh](../../policies/mesh) policy.

> Additionally, when running on Universal you have to ensure that every data plane in the mesh has been configured with a Dataplane Token.

Reviewer: ```suggestion
Additionally, when running on Universal you have to ensure that every data plane in the mesh has been configured with a data plane Token.
```

Author: I think it should be either `data plane token` or `Dataplane Token`, because it is our termⓇ™️

Reviewer 2: @jakubdyszkiewicz data plane proxy token

> When mTLS is enabled, every connection between dataplanes is denied by default, so you have to explicitly allow it using [TrafficPermission](../../policies/traffic-permissions).

Reviewer: ```suggestion
When mTLS is enabled, every connection between data planes is denied by default, so you have to explicitly allow it using [TrafficPermission](../../policies/traffic-permissions).
```

Kuma also supports other CAs. For further details refer to [Mutual TLS](../../policies/mutual-tls) policy.

## Data plane to control plane communication

A data plane connects to the control plane for its configuration, including mTLS certificates described in the previous sections.

### Encrypted communication

Because the data plane and the control plane exchange sensitive information, the communication needs to be encrypted by TLS.

> The control plane's server that is consumed by the data plane is secured by TLS by default with autogenerated certificates.

Reviewer: ```suggestion
By default, the control plane's server that is consumed by the data plane is secured by TLS with autogenerated certificates.
```

> For the maximum security, it is recommended that the data plane verifies the identity of the control plane, therefore it needs to obtain the CA that was used to generate the certificate by which the control plane's server is secured.

Reviewer: ```suggestion
For security purposes, the data plane must verify the identity of the control plane. To do so, data planes needs to obtain the CA that was used to generate the certificate by which the control plane's server is secured.
```

:::
::::

#### Multizone

> When running in multizone mode you can generate tokens both on Global and Remote Control Plane

Reviewer: ```suggestion
When running in multizone mode, you can generate tokens both on Global and Remote Control Plane
```

> If you configure your deployment pipeline to generate Dataplane Token before running a dataplane you can use Remote CP. This way Global CP is not a single point of failure.

Reviewer: ```suggestion
If you configure your deployment pipeline to generate data plane tokens before running a data plane, you can use Remote CP. This way Global CP is not a single point of failure.
```
Would like this to be slightly expanded on since I am slightly confused. May cause confusion for external folks too.

Author: To be honest I'm not sure how to extend this section 🤔 
If you have a setup that you generate dp token before you run the kuma-dp then if you do this on global and global is down, you cannot spawn a new DP. If you do this on remote and global is down you are still fine with spawning new dps
